app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -452,6 +495,46 @@ export function testsForRpcMethodAssumingNoBlockParam(
     });
   });
 
+  for (const paramIndex of [...Array(numberOfParameters).keys()]) {
+    it(`does not reuse the result of a previous request if parameter at index "${paramIndex}" differs`, async () => {
```

Comment:
took me a while to work through why this was testing what you claim its testing. But I see now! Looks good!

Reply:
Yeah... the param builder function didn't quite do what I wanted here, so I constructed the parameters inline in the test. I should probably find a way to make this more readable.

Reply:
Would you mind explaining which part you initially found confusing? That might help in finding a clearer way to write this.

Reply:
I don't think there's any issue with the description. I think it was just towards the end of my day and my brain wasn't working as well. So I guess this comment was more for me and not very generalizable 😅